src/Controls/src/Core/Platform/Android/Extensions/SemanticExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿#nullable enable
+
+using AndroidX.Core.View;
+using AndroidX.Core.View.Accessibility;
+using AView = Android.Views.View;
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	public static partial class SemanticExtensions
+	{
+		public static void UpdateSemanticNodeInfo(this View virtualView, AccessibilityNodeInfoCompat? info)
+		{
+			if (info == null)
+				return;
+
+			if(virtualView.TapGestureRecognizerNeedsDelegate())
+				info.AddAction(AccessibilityNodeInfoCompat.AccessibilityActionCompat.ActionClick);
+		}
+
+		internal static void ApplyControlsAccessibilityDelegateIfNeeded(this View virtualView)
+		{
+			if (virtualView?.Handler?.NativeView is not AView view)
+				return;
+
+			bool needsDelegate = virtualView.TapGestureRecognizerNeedsDelegate();
```

Comment:
should we separate out the TapGestureRecognizer logic? not sure if I'm misunderstood, but I see this method/PR as being primarily about the ability to access/customize the nodeInfo and accessibilityDelegate, and secondarily about the gesture stuff